### ISSUE TITLE
containsElements("H2O", "Z") returns TRUE

### ISSUE THREAD

**Issue by sgibb** [rformassspectrometry/MetaboCoreUtils#63]
`containsElements` returns `TRUE` (with a warning) for characters that are not valid elements:

``` r
library("MetaboCoreUtils")

containsElements("H2O", "Z")
#> Warning in (function (xx, rr) : The following names are not valid and are
#> dropped: 1
#> [1] TRUE
containsElements("H2O", "J")
#> Warning in (function (xx, rr) : The following names are not valid and are
#> dropped: 1
#> [1] TRUE
containsElements("H2O", "")
#> Warning in (function (xx, rr) : The following names are not valid and are
#> dropped: 1
#> [1] TRUE
```

<sup>Created on 2023-08-02 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

It should return `FALSE` (without a warning).

It would be possible to add something like 

```r
if (length(rr) == 1 && rr < 0)
    return(integer(0))
```

in the `mapply` of `countElements` https://github.com/rformassspectrometry/MetaboCoreUtils/blob/7294eecf45a46e873c296edee8eb5e4af3c04ea0/R/chemFormula.R#L55-L75 (after the `if (is.na(xx))` statement introduced in #62 ) and test for `length(x) == 0` in `containsElements` but that would break many unit tests in `test_adducts.R` (`.pasteElements` throws an error because of an unnamed argument instead of silently return `""`; even returning a named `integer(0)` results in many other failures in `test_adducts.R`).

**Comment by sgibb** [comment]
I add (currently failing) unit tests in the [issue63-containsElements-returns-TRUE-for-invalid-elements](https://github.com/rformassspectrometry/MetaboCoreUtils/tree/issue63-containsElements-returns-TRUE-for-invalid-elements)
https://github.com/rformassspectrometry/MetaboCoreUtils/compare/issue63-containsElements-returns-TRUE-for-invalid-elements?expand=1

**Comment by jorainer** [comment]
Thanks for adding this issue! Can you maybe work on that?

**Comment by sgibb** [comment]
fixed by #65
